File: flare-web-client/src/flareclient-extensions-builtin/tasks/force-logout/force-logout-task.js
// Delegates to FlareClient's own _handleAuthFailure() — the one place
// logout logic lives on web. Never reimplement it here.
export const ForceLogoutTask = {
  id: "force_logout",
  execute(_params) {
    if (window.__flareClient__) {
      window.__flareClient__._handleAuthFailure();
    } else {
      console.warn("[ForceLogoutTask] no active FlareClient instance found");
    }
  }
};

File: flare-android-client/flareclient-extensions-builtin/src/main/java/dev/flareframework/extensions/builtin/tasks/force_logout/ForceLogoutTask.java
package dev.flareframework.extensions.builtin.tasks.force_logout;

import android.app.Activity;
import dev.flareframework.client.FlareClientActivity;
import dev.flareframework.client.flare.task.FlareClientTask;
import org.json.JSONObject;

/**
 * Delegates to FlareClientActivity.clearStorage() — the ONE place logout
 * logic lives. Never reimplement token-clearing/finish() here.
 */
public class ForceLogoutTask implements FlareClientTask {
    public static final String ID = "force_logout";

    @Override
    public String id() { return ID; }

    @Override
    public void execute(Activity host, JSONObject params) {
        if (host instanceof FlareClientActivity) {
            ((FlareClientActivity) host).clearStorage();
        }
    }
}